--- xulrunner-16.0.1/mozilla/embedding/browser/gtk/src/EmbedPrivate.h~	2012-10-23 21:47:20.819022685 +0200
+++ xulrunner-16.0.1/mozilla/embedding/browser/gtk/src/EmbedPrivate.h	2012-10-23 23:22:14.899108295 +0200
@@ -147,7 +147,7 @@
   // the list of all open windows
   static nsTArray<EmbedPrivate*> *sWindowList;
   // what is our profile path?
-  static nsILocalFile           *sProfileDir;
+  static nsIFile                *sProfileDir;
   static nsISupports            *sProfileLock;
 
   static nsIDirectoryServiceProvider * sAppFileLocProvider;
--- xulrunner-16.0.1/mozilla/embedding/browser/gtk/src/EmbedPrivate.cpp~	2012-10-23 21:47:20.819022685 +0200
+++ xulrunner-16.0.1/mozilla/embedding/browser/gtk/src/EmbedPrivate.cpp	2012-10-23 23:22:40.842440716 +0200
@@ -89,7 +89,7 @@
 char                    *EmbedPrivate::sPath        = nsnull;
 char                    *EmbedPrivate::sCompPath    = nsnull;
 nsTArray<EmbedPrivate*> *EmbedPrivate::sWindowList  = nsnull;
-nsILocalFile            *EmbedPrivate::sProfileDir  = nsnull;
+nsIFile                 *EmbedPrivate::sProfileDir  = nsnull;
 nsISupports             *EmbedPrivate::sProfileLock = nsnull;
 GtkWidget               *EmbedPrivate::sOffscreenWindow = 0;
 GtkWidget               *EmbedPrivate::sOffscreenFixed  = 0;
@@ -507,7 +507,7 @@
   if (sWidgetCount == 1) {
     nsresult rv;
 
-    nsCOMPtr<nsILocalFile> binDir;
+    nsCOMPtr<nsIFile> binDir;
     if (sCompPath) {
       rv = NS_NewNativeLocalFile(nsDependentCString(sCompPath), 1, getter_AddRefs(binDir));
       if (NS_FAILED(rv))
@@ -522,7 +522,7 @@
     if (!grePath)
       return;
 
-    nsCOMPtr<nsILocalFile> greDir;
+    nsCOMPtr<nsIFile> greDir;
     rv = NS_NewNativeLocalFile(nsDependentCString(grePath), PR_TRUE,
                                getter_AddRefs(greDir));
     if (NS_FAILED(rv))
--- xulrunner-16.0.1/mozilla/embedding/browser/gtk/src/EmbedWindow.cpp~	2012-10-23 21:47:20.819022685 +0200
+++ xulrunner-16.0.1/mozilla/embedding/browser/gtk/src/EmbedWindow.cpp	2012-10-23 23:51:30.099046377 +0200
@@ -472,3 +472,12 @@
 
   return rv;
 }
+
+NS_IMETHODIMP
+EmbedWindow::Blur()
+{
+  NS_NOTREACHED("EmbedWindow::Blur not supported in EmbedWindow");
+
+  return NS_ERROR_NOT_IMPLEMENTED;
+}
+
