--- mozilla/config/autoconf.mk.in.org	2007-01-07 04:00:25.140001000 +0200
+++ mozilla/config/autoconf.mk.in	2007-01-07 04:00:40.140338541 +0200
@@ -63,7 +63,7 @@
 mandir		= @mandir@
 idldir		= @datadir@/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 
-mozappdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+mozappdir	= $(libdir)/$(MOZ_APP_NAME)
 mredir		= $(libdir)/mre/mre-$(MOZ_APP_VERSION)
 mrelibdir	= $(mredir)/lib
 
--- mozilla/configure.in~	2007-01-07 02:50:05.085718876 +0200
+++ mozilla/configure.in	2007-01-07 04:03:09.003684862 +0200
@@ -801,7 +801,7 @@
 
 MOZ_JS_LIBS='-L$(DIST)/bin -lmozjs'
 DYNAMIC_XPCOM_LIBS='-L$(DIST)/bin -lxpcom -lxpcom_core'
-MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(DIST)/bin'
+MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(DIST)/bin -Wl,-rpath,$(mozappdir)'
 XPCOM_FROZEN_LDOPTS='-L$(DIST)/bin $(MOZ_FIX_LINK_PATHS) -lxpcom'
 LIBXUL_LIBS='$(XPCOM_FROZEN_LDOPTS) -lxul'
 XPCOM_GLUE_LDOPTS='$(DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) $(XPCOM_FROZEN_LDOPTS)'
--- mozilla/configure~	2007-01-07 04:22:18.799552498 +0200
+++ mozilla/configure	2007-01-07 04:22:32.199854444 +0200
@@ -5032,7 +5032,7 @@
 
 MOZ_JS_LIBS='-L$(DIST)/bin -lmozjs'
 DYNAMIC_XPCOM_LIBS='-L$(DIST)/bin -lxpcom -lxpcom_core'
-MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(DIST)/bin'
+MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(DIST)/bin -Wl,-rpath,$(mozappdir)'
 XPCOM_FROZEN_LDOPTS='-L$(DIST)/bin $(MOZ_FIX_LINK_PATHS) -lxpcom'
 LIBXUL_LIBS='$(XPCOM_FROZEN_LDOPTS) -lxul'
 XPCOM_GLUE_LDOPTS='$(DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) $(XPCOM_FROZEN_LDOPTS)'
