--- xulrunner-7.0/mozilla/toolkit/mozapps/installer/packager.mk.orig	2011-09-22 22:20:42.000000000 +0200
+++ xulrunner-7.0/mozilla/toolkit/mozapps/installer/packager.mk	2011-09-28 11:14:05.247518747 +0200
@@ -632,7 +632,7 @@
 	@$(NSINSTALL) -D $(DIST)/xpt
 	@($(XPIDL_LINK) $(DIST)/xpt/$(MOZ_PKG_APPNAME).xpt $(DIST)/$(STAGEPATH)$(MOZ_PKG_DIR)$(_BINPATH)/components/*.xpt && rm -f $(DIST)/$(STAGEPATH)$(MOZ_PKG_DIR)$(_BINPATH)/components/*.xpt && cp $(DIST)/xpt/$(MOZ_PKG_APPNAME).xpt $(DIST)/$(STAGEPATH)$(MOZ_PKG_DIR)$(_BINPATH)/components && printf "interfaces $(MOZ_PKG_APPNAME).xpt\n" >$(DIST)/$(STAGEPATH)$(MOZ_PKG_DIR)$(_BINPATH)/components/interfaces.manifest) || echo No *.xpt files found in: $(DIST)/$(STAGEPATH)$(MOZ_PKG_DIR)$(_BINPATH)/components/.  Continuing...
 else
-	@cd $(DIST)/bin && $(TAR) $(TAR_CREATE_FLAGS) - * | (cd ../$(MOZ_PKG_DIR); tar -xf -)
+	cp -rfLp $(DIST)/bin/* $(MOZ_PKG_DIR)
 endif
 else
 	@cd $(DIST)/bin && tar $(TAR_CREATE_FLAGS) - * | (cd ../$(MOZ_PKG_DIR); tar -xf -)
@@ -725,27 +725,30 @@
 	$(error "make install" is not supported on this platform. Use "make package" instead.)
 endif
 	$(NSINSTALL) -D $(DESTDIR)$(installdir)
-	(cd $(DIST)/$(MOZ_PKG_DIR) && $(TAR) --exclude=precomplete $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(installdir) && tar -xf -)
+	cp -rfLp $(DIST)/$(MOZ_PKG_DIR)/* $(DESTDIR)$(installdir)
 	$(NSINSTALL) -D $(DESTDIR)$(bindir)
 	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
-	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
+	ln -s $(installdir)/$(MOZ_APP_NAME)-bin $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
 ifdef INSTALL_SDK # Here comes the hard part
 	$(NSINSTALL) -D $(DESTDIR)$(includedir)
-	(cd $(DIST)/include && $(TAR) $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(includedir) && tar -xf -)
+	cp -rfLp $(DIST)/include/* $(DESTDIR)$(includedir)
 	$(NSINSTALL) -D $(DESTDIR)$(idldir)
-	(cd $(DIST)/idl && $(TAR) $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(idldir) && tar -xf -)
+	cp -rfLp $(DIST)/idl/* $(DESTDIR)$(idldir)
 # SDK directory is the libs + a bunch of symlinks
 	$(NSINSTALL) -D $(DESTDIR)$(sdkdir)/sdk/lib
 	$(NSINSTALL) -D $(DESTDIR)$(sdkdir)/sdk/bin
 	if test -f $(DIST)/include/xpcom-config.h; then \
 	  $(SYSINSTALL) $(IFLAGS1) $(DIST)/include/xpcom-config.h $(DESTDIR)$(sdkdir); \
 	fi
 	find $(DIST)/sdk -name "*.pyc" | xargs rm -f
-	(cd $(DIST)/sdk/lib && $(TAR) $(TAR_CREATE_FLAGS) - .) | (cd $(DESTDIR)$(sdkdir)/sdk/lib && tar -xf -)
-	(cd $(DIST)/sdk/bin && $(TAR) $(TAR_CREATE_FLAGS) - .) | (cd $(DESTDIR)$(sdkdir)/sdk/bin && tar -xf -)
+	cp -rfLp $(DIST)/sdk/lib/* $(DESTDIR)$(sdkdir)/sdk/lib
+	cp -rfLp $(DIST)/sdk/bin/* $(DESTDIR)$(sdkdir)/sdk/bin
+	# replace copies with symlinks
+	ln -sf $(installdir)/libmozjs.so $(DESTDIR)$(sdkdir)/sdk/lib/libmozjs.so
+	ln -sf $(installdir)/libxpcom.so $(DESTDIR)$(sdkdir)/sdk/lib/libxpcom.so
+	ln -sf $(installdir)/libxul.so $(DESTDIR)$(sdkdir)/sdk/lib/libxul.so
+	ln -sf $(installdir)/libmozalloc.so $(DESTDIR)$(sdkdir)/sdk/lib/libmozalloc.so
+	chmod a+rx $(DESTDIR)$(sdkdir)/sdk/bin/xpt.py # temp fix for https://bugzilla.mozilla.org/show_bug.cgi?id=63955
 	$(RM) -f $(DESTDIR)$(sdkdir)/lib $(DESTDIR)$(sdkdir)/bin $(DESTDIR)$(sdkdir)/include $(DESTDIR)$(sdkdir)/include $(DESTDIR)$(sdkdir)/sdk/idl $(DESTDIR)$(sdkdir)/idl
 	ln -s $(sdkdir)/sdk/lib $(DESTDIR)$(sdkdir)/lib
 	ln -s $(installdir) $(DESTDIR)$(sdkdir)/bin
