--- seamonkey-1.0/configure.in.orig	2006-01-10 04:11:42.000000000 +0100
+++ seamonkey-1.0/configure.in	2006-02-01 21:10:12.670658750 +0100
@@ -7098,7 +7098,7 @@
         mkdir "directory/c-sdk/ldap"    
     fi
 
-    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
+    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-nss --with-mozilla"
     if test -z "$MOZ_DEBUG"; then
         ac_configure_args="$ac_configure_args --disable-debug"
     fi
--- seamonkey-1.0/configure.orig	2006-01-11 06:43:03.000000000 +0100
+++ seamonkey-1.0/configure	2006-02-01 21:10:17.982990750 +0100
@@ -20209,7 +20209,7 @@
         mkdir "directory/c-sdk/ldap"    
     fi
 
-    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
+    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-nss --with-mozilla"
     if test -z "$MOZ_DEBUG"; then
         ac_configure_args="$ac_configure_args --disable-debug"
     fi
--- mozilla/directory/c-sdk/ldap/libraries/libssldap/Makefile.in.orig	Fri May 17 08:54:00 2002
+++ mozilla/directory/c-sdk/ldap/libraries/libssldap/Makefile.in	Wed Aug 28 11:37:29 2002
@@ -53,9 +53,8 @@
 LOCAL_INCLUDES  = -I$(PUBLIC)/nspr
 INCLUDES	+= -I$(DISTHDIR) -I$(HDIR) -I$(INSTALLDIR)/include \
 		   -I$(DIST)/include \
-		   -I$(DIST)/public/security \
-		   -I$(DIST)/public/nss \
-                   -I$(DIST)/include/nspr
+		   -I/usr/include/nss \
+                   -I/usr/include/nspr
 DEFINES		+= $(DEFS)
 
 PLATFORMCFLAGS	= -DUSE_WAITPID -DNEEDPROTOS
