diff -ur mozilla.orig/toolkit/content/about.xhtml mozilla/toolkit/content/about.xhtml
--- mozilla.orig/toolkit/content/about.xhtml	2008-03-31 22:13:37.000000000 +0200
+++ mozilla/toolkit/content/about.xhtml	2008-06-22 15:44:55.000000000 +0200
@@ -91,7 +91,7 @@
                                  .getService(Components.interfaces.nsIXULAppInfo)
                                  .version;
       var version = document.getElementById("version");
-      version.appendChild(document.createTextNode("&about.version; " + versionNum));
+      version.appendChild(document.createTextNode("XULRunner " + versionNum));
 
       // append user agent
       var ua = navigator.userAgent;
--- mozilla/xulrunner/app/xulrunner.js.orig	2007-09-05 15:47:24.000000000 +0200
+++ mozilla/xulrunner/app/xulrunner.js	2008-07-23 13:57:13.000000000 +0200
@@ -69,3 +69,5 @@
 #else
 pref("browser.preferences.animateFadeIn", false);
 #endif
+pref("browser.xul.error_pages.expert_bad_cert", true);
+pref("browser.ssl_override_behavior", 2);
