diff -ur mozilla.orig/toolkit/content/about.xhtml mozilla/toolkit/content/about.xhtml
--- mozilla.orig/toolkit/content/about.xhtml	2008-03-31 22:13:37.000000000 +0200
+++ mozilla/toolkit/content/about.xhtml	2008-06-22 15:44:55.000000000 +0200
@@ -62,7 +62,7 @@
   <div id="aboutLogoContainer">
     <a id="vendorURL" href="http://www.mozilla.org/">
       <img src="about:logo" alt="&brandShortName;"/>
-#expand <p id="version">&about.version; __MOZ_APP_VERSION__</p>
+#expand <p id="version">XULRunner __MOZ_APP_VERSION__</p>
     </a>
   </div>
 
--- mozilla/xulrunner/app/xulrunner.js.orig	2007-09-05 15:47:24.000000000 +0200
+++ mozilla/xulrunner/app/xulrunner.js	2008-07-23 13:57:13.000000000 +0200
@@ -69,3 +69,5 @@
 #else
 pref("browser.preferences.animateFadeIn", false);
 #endif
+pref("browser.xul.error_pages.expert_bad_cert", true);
+pref("browser.ssl_override_behavior", 2);
