--- mozilla-2.0/configure.in	2011-03-19 00:33:33.000000000 +0100
+++ mozilla-2.0.new/configure.in	2011-03-23 16:18:45.746667659 +0100
@@ -960,7 +960,7 @@ TARGET_MD_ARCH=unix
 DIRENT_INO=d_ino
 MOZ_USER_DIR=".mozilla"
 
-MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin -Wl,-rpath-link,$(prefix)/lib'
+MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin -Wl,-rpath-link,$(prefix)/lib -Wl,-rpath,$(installdir)'
 
 MOZ_FS_LAYOUT=unix
 
--- mozilla-2.0/configure~	2014-11-26 03:18:44.000000000 +0100
+++ mozilla-2.0/configure	2014-12-01 23:14:34.554999919 +0100
@@ -6473,7 +6473,7 @@ TARGET_MD_ARCH=unix
 DIRENT_INO=d_ino
 MOZ_USER_DIR=".mozilla"
 
-MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin -Wl,-rpath-link,$(prefix)/lib'
+MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin -Wl,-rpath-link,$(prefix)/lib -Wl,-rpath,$(installdir)'
 
 MOZ_FS_LAYOUT=unix
 

