--- mozilla/build/unix/Makefile~	2007-01-07 01:38:09.839104485 +0200
+++ mozilla/build/unix/Makefile	2007-01-07 01:38:17.189270040 +0200
@@ -77,9 +77,9 @@
 	cat $< | sed \
 	-e "s|%prefix%|$(prefix)|" \
 	-e "s|%exec_prefix%|$(exec_prefix)|" \
-	-e "s|%includedir%|$(includedir)|" \
-	-e "s|%libdir%|$(mozappdir)|" \
-	-e "s|%idldir%|$(idldir)|" \
+	-e "s|%includedir%|/usr/include/$(MOZ_APP_NAME)|" \
+	-e "s|%libdir%|$(libdir)/$(MOZ_APP_NAME)|" \
+	-e "s|%idldir%|/usr/share/idl/$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%DEFS%|$(_DEFS)|" \
 	-e "s|%FULL_NSPR_LIBS%|$(FULL_NSPR_LIBS)|" \
@@ -89,9 +89,9 @@
 	cat $< | sed \
 	-e "s|%prefix%|$(prefix)|" \
 	-e "s|%exec_prefix%|$(exec_prefix)|" \
-	-e "s|%includedir%|$(includedir)|" \
-	-e "s|%libdir%|$(mozappdir)|" \
-	-e "s|%idldir%|$(idldir)|" \
+	-e "s|%includedir%|/usr/include/$(MOZ_APP_NAME)|" \
+	-e "s|%libdir%|$(libdir)/$(MOZ_APP_NAME)|" \
+	-e "s|%idldir%|/usr/share/idl/$(MOZ_APP_NAME)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%SUPERWIN_LIBS%|$(SUPERWIN_LIBS)|" \
--- mozilla/build/unix/mozilla-plugin.pc.in~	2005-04-22 20:11:13.000000000 +0300
+++ mozilla/build/unix/mozilla-plugin.pc.in	2007-01-07 01:29:41.427650511 +0200
@@ -7,4 +7,4 @@
 Description: Mozilla Plug-In API
 Version: %MOZILLA_VERSION%
 Requires: %MOZ_APP_NAME%-xpcom = %MOZILLA_VERSION%
-Cflags: -I${includedir}/java -I${includedir}/plugin
+Cflags: -I${includedir}/java -I${includedir}/plugin -I${includedir}/dom
