--- mozilla/configure~	2008-01-29 19:51:03.000000000 +0000
+++ mozilla/configure	2008-03-06 20:24:20.820277911 +0000
@@ -19729,7 +19729,7 @@
 fi
 
 if test "$MOZ_LDAP_XPCOM"; then
-    LDAP_CFLAGS='-I${DIST}/public/ldap'
+    LDAP_CFLAGS=''
     if test "$OS_ARCH" = "WINNT"; then
         if test -n "$GNU_CC"; then
             LDAP_LIBS='-L$(DIST)/lib -lnsldap32v60 -lnsldappr32v60 -lnsldif32v60'
@@ -19739,7 +19739,7 @@
     elif test "$OS_ARCH" = "OS2"; then
             LDAP_LIBS='$(DIST)/lib/$(LIB_PREFIX)ldap60.${IMPORT_LIB_SUFFIX} $(DIST)/lib/$(LIB_PREFIX)prldap60.${IMPORT_LIB_SUFFIX} $(DIST)/lib/$(LIB_PREFIX)ldif60.${IMPORT_LIB_SUFFIX}'
     else
-        LDAP_LIBS='-L${DIST}/bin -L${DIST}/lib -lldap60 -lprldap60 -lldif60'
+        LDAP_LIBS='-lldap60 -lprldap60 -lldif60'
     fi
 fi
 
@@ -21178,7 +21178,7 @@
 unset CONFIG_FILES
 
 if test "$COMPILE_ENVIRONMENT"; then
-if test -z "$MOZ_NATIVE_NSPR" || test "$MOZ_LDAP_XPCOM"; then
+if test -z "$MOZ_NATIVE_NSPR"; then
     ac_configure_args="$_SUBDIR_CONFIG_ARGS --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-mozilla"
     if test -z "$MOZ_DEBUG"; then
         ac_configure_args="$ac_configure_args --disable-debug"
@@ -21311,20 +21311,6 @@
 # the c-sdk first.  
 #
 if test "$MOZ_LDAP_XPCOM"; then
-
-    # these subdirs may not yet have been created in the build tree.
-    # don't use the "-p" switch to mkdir, since not all platforms have it
-    #
-    if test ! -d "directory"; then
-        mkdir "directory"
-    fi
-    if test ! -d "directory/c-sdk"; then
-        mkdir "directory/c-sdk"    
-    fi
-    if test ! -d "directory/c-sdk/ldap"; then
-        mkdir "directory/c-sdk/ldap"    
-    fi
-
     ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
     if test -z "$MOZ_DEBUG"; then
         ac_configure_args="$ac_configure_args --disable-debug"
--- mozilla/toolkit/toolkit-tiers.mk~	2007-07-23 22:02:20.000000000 +0000
+++ mozilla/toolkit/toolkit-tiers.mk	2008-03-06 21:35:08.892538703 +0000
@@ -260,11 +260,6 @@
 tier_toolkit_dirs += tools/trace-malloc
 endif
 
-ifdef MOZ_LDAP_XPCOM
-tier_toolkit_staticdirs += directory/c-sdk
-tier_toolkit_dirs	+= directory/xpcom
-endif
-
 ifdef MOZ_ENABLE_GNOME_COMPONENT
 tier_toolkit_dirs    += toolkit/system/gnome
 endif
