--- mozilla/toolkit/mozapps/installer/packager.mk.orig	2008-04-28 21:14:12.000000000 +0200
+++ mozilla/toolkit/mozapps/installer/packager.mk	2008-06-20 23:51:52.000000000 +0200
@@ -339,13 +339,13 @@
 # NOTE: this must be a tar now that dist links into the tree so that we
 # do not strip the binaries actually in the tree.
 	@echo "Creating package directory..."
-	@mkdir $(DIST)/$(MOZ_PKG_APPNAME)
+	@mkdir -p $(DESTDIR)$(MOZ_PKG_APPDIR)
 ifdef MOZ_PKG_MANIFEST
 	$(RM) -rf $(DIST)/xpt
 	$(call PACKAGER_COPY, "$(DIST)",\
-		 "$(DIST)/$(MOZ_PKG_APPNAME)", \
+		 "$(DESTDIR)/$(MOZ_PKG_APPDIR)", \
 		"$(MOZ_PKG_MANIFEST)", "$(PKGCP_OS)", 1, 0, 1)
-	$(PERL) $(topsrcdir)/xpinstall/packager/xptlink.pl -s $(DIST) -d $(DIST)/xpt -f $(DIST)/$(MOZ_PKG_APPNAME)/components -v -x "$(XPIDL_LINK)"
+	$(PERL) $(topsrcdir)/xpinstall/packager/xptlink.pl -s $(DIST) -d $(DIST)/xpt -f $(DESTDIR)/$(MOZ_PKG_APPDIR)/components -v -x "$(XPIDL_LINK)"
 else # !MOZ_PKG_MANIFEST
 ifeq ($(MOZ_PKG_FORMAT),DMG)
 # If UNIVERSAL_BINARY, the package will be made from an already-prepared
@@ -358,7 +358,7 @@
 endif
 endif
 else
-	@cd $(DIST)/bin && tar $(TAR_CREATE_FLAGS) - * | (cd ../$(MOZ_PKG_APPNAME); tar -xf -)
+	cp -rfLp $(DIST)/bin/* $(DESTDIR)$(MOZ_PKG_APPDIR)
 endif # DMG
 endif # MOZ_PKG_MANIFEST
 ifndef PKG_SKIP_STRIP
@@ -390,7 +390,7 @@
 endif
 	@echo "Removing unpackaged files..."
 ifdef NO_PKG_FILES
-	cd $(DIST)/$(STAGEPATH)$(MOZ_PKG_APPNAME)$(_BINPATH); rm -rf $(NO_PKG_FILES)
+	cd $(DESTDIR)$(MOZ_PKG_APPDIR)$(_BINPATH); rm -rf $(NO_PKG_FILES)
 endif
 ifdef MOZ_PKG_REMOVALS
 	$(SYSINSTALL) $(MOZ_PKG_REMOVALS_GEN) $(DIST)/$(STAGEPATH)$(MOZ_PKG_APPNAME)$(_BINPATH)
@@ -414,39 +414,47 @@
 	$(error "make install" is not supported on this platform. Use "make package" instead.)
 endif
 	$(NSINSTALL) -D $(DESTDIR)$(installdir)
-	(cd $(DIST)/$(MOZ_PKG_APPNAME) && tar $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(installdir) && tar -xf -)
 	$(NSINSTALL) -D $(DESTDIR)$(bindir)
 	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
-	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
+	ln -s $(installdir)/$(MOZ_APP_NAME)-bin $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
+	$(RM) -f $(DESTDIR)$(bindir)/regxpcom
+	ln -s $(installdir)/regxpcom $(DESTDIR)$(bindir)/regxpcom
+	$(RM) -f $(DESTDIR)$(bindir)/xpcshell
+	ln -s $(installdir)/xpcshell $(DESTDIR)$(bindir)/xpcshell
+	$(RM) -f $(DESTDIR)$(bindir)/xpidl
+	ln -s $(installdir)/xpidl $(DESTDIR)$(bindir)/xpidl
+	$(RM) -f $(DESTDIR)$(bindir)/xpt_dump
+	ln -s $(installdir)/xpt_dump $(DESTDIR)$(bindir)/xpt_dump
+	$(RM) -f $(DESTDIR)$(bindir)/xpt_link
+	ln -s $(installdir)/xpt_link $(DESTDIR)$(bindir)/xpt_link
 ifdef INSTALL_SDK # Here comes the hard part
 # include directory is stable (dist/sdk/include) and unstable (dist/include)
 	$(NSINSTALL) -D $(DESTDIR)$(includedir)/stable
-	$(NSINSTALL) -D $(DESTDIR)$(includedir)/unstable
-	(cd $(DIST)/sdk/include && tar $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(includedir)/stable && tar -xf -)
+	cp -rfLp $(DIST)/sdk/include/* $(DESTDIR)$(includedir)/stable
 # The dist/include has module subdirectories that we need to flatten
-	find $(DIST)/include -xtype f -exec $(SYSINSTALL) $(IFLAGS1) {} $(DESTDIR)$(includedir)/unstable \;
+#	find $(DIST)/include -xtype f -exec $(SYSINSTALL) $(IFLAGS1) {} $(DESTDIR)$(includedir) \;
+	cp -rfLp $(DIST)/include/* $(DESTDIR)$(includedir)
 # IDL directory is stable (dist/sdk/idl) and unstable (dist/idl)
 	$(NSINSTALL) -D $(DESTDIR)$(idldir)/stable 
-	$(NSINSTALL) -D $(DESTDIR)$(idldir)/unstable
-	(cd $(DIST)/sdk/idl && tar $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(idldir)/stable && tar -xf -)
-	(cd $(DIST)/idl && tar $(TAR_CREATE_FLAGS) - .) | \
-	  (cd $(DESTDIR)$(idldir)/unstable && tar -xf -)
+	$(NSINSTALL) -D $(DESTDIR)$(idldir)
+	cp -rfLp $(DIST)/sdk/idl/* $(DESTDIR)$(idldir)/stable
+	cp -rfLp $(DIST)/idl/* $(DESTDIR)$(idldir)
 # SDK directory is the libs + a bunch of symlinks
 	$(NSINSTALL) -D $(DESTDIR)$(sdkdir)/sdk/lib
 	if test -f $(DIST)/sdk/include/xpcom-config.h; then \
 	  $(SYSINSTALL) $(IFLAGS1) $(DIST)/sdk/include/xpcom-config.h $(DESTDIR)$(sdkdir); \
 	fi
-	(cd $(DIST)/sdk/lib && tar $(TAR_CREATE_FLAGS) - .) | (cd $(DESTDIR)$(sdkdir)/sdk/lib && tar -xf -)
+	cp -rfLp $(DIST)/sdk/lib/*.a $(DESTDIR)$(sdkdir)/sdk/lib
+	ln -s $(installdir)/libmozjs.so $(DESTDIR)$(sdkdir)/sdk/lib/libmozjs.so
+	ln -s $(installdir)/libxpcom.so $(DESTDIR)$(sdkdir)/sdk/lib/libxpcom.so
+	ln -s $(installdir)/libxul.so $(DESTDIR)$(sdkdir)/sdk/lib/libxul.so
 	$(RM) -f $(DESTDIR)$(sdkdir)/lib $(DESTDIR)$(sdkdir)/bin $(DESTDIR)$(sdkdir)/sdk/include $(DESTDIR)$(sdkdir)/include $(DESTDIR)$(sdkdir)/sdk/idl $(DESTDIR)$(sdkdir)/idl
 	ln -s $(sdkdir)/sdk/lib $(DESTDIR)$(sdkdir)/lib
 	ln -s $(installdir) $(DESTDIR)$(sdkdir)/bin
 	ln -s $(includedir)/stable $(DESTDIR)$(sdkdir)/sdk/include
-	ln -s $(includedir)/unstable $(DESTDIR)$(sdkdir)/include
+	ln -s $(includedir) $(DESTDIR)$(sdkdir)/include
 	ln -s $(idldir)/stable $(DESTDIR)$(sdkdir)/sdk/idl
-	ln -s $(idldir)/unstable $(DESTDIR)$(sdkdir)/idl
+	ln -s $(idldir) $(DESTDIR)$(sdkdir)/idl
 endif # INSTALL_SDK
 
 make-sdk:
